todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskPresenter.java

```diff
@@ -73,7 +76,7 @@ void setupListeners() {
 
     @Override
     public void start() {
-        if (!isNewTask()) {
+        if (!isNewTask() && mIsDataMissing) {
```

Comment:
I feel like this logic should be in the `Repository` and not in the `Presenter`. How about, instead of passing in data through the `Bundle`, we use `SharedPreferences` to store wether the data needs refresh or not. That way, exposing that information through a `Provider`, the `Repository` can ask the `Provider` if a refresh is needed and trigger it. It can also be tested in the `Repository`, which always makes a puppy happy.